app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script.rb

```diff
@@ -33,6 +33,37 @@ def create_in_provider_queue(manager_id, params, auth_user = nil)
 
       MiqTask.generic_action_with_callback(task_opts, queue_opts)
     end
+
+    def update_in_provider(manager_id, params)
```

Comment:
You can definitely combine update and create methods into one common method, same to the queue version.

The public methods you can still have create and update separated, but they redirect to a common private method.